File: MobilneSystemyInformatyczne/ShareWindsurfingEquipment/src/main/java/com/project/ShareWindsurfingEquipment/model/Equipment.java
package com.project.ShareWindsurfingEquipment.model;

import com.project.ShareWindsurfingEquipment.common.EquipmentType;

import javax.persistence.*;
import java.math.BigDecimal;

@Entity
@Table
public class Equipment {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column
    private String brand;

    @Column
    private String model;

    @Column
    private String urlImage;

    @Column
    private BigDecimal pricePerHour;

    @Column
    private String lenderInstructor;

    @Column
    private String size;

    @Column
    private String description;

    @Column
    private int year;

    @Enumerated(EnumType.STRING)
    @Column
    private EquipmentType equipmentType;

    public Equipment() {
    }

    public Equipment(String brand, String model, String urlImage, BigDecimal pricePerHour, String lenderInstructor, String size, String description, int year, String equipmentType) {
        this.brand = brand;
        this.model = model;
        this.urlImage = urlImage;
        this.pricePerHour = pricePerHour;
        this.lenderInstructor = lenderInstructor;
        this.size = size;
        this.description = description;
        this.year = year;
        this.equipmentType = EquipmentType.valueOf(equipmentType);
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getBrand() {
        return brand;
    }

    public void setBrand(String brand) {
        this.brand = brand;
    }

    public BigDecimal getPricePerHour() {
        return pricePerHour;
    }

    public void setPricePerHour(BigDecimal pricePerHour) {
        this.pricePerHour = pricePerHour;
    }

    public String getLenderInstructor() {
        return lenderInstructor;
    }

    public void setLenderInstructor(String lenderInstructor) {
        this.lenderInstructor = lenderInstructor;
    }

    public String getSize() {
        return size;
    }

    public void setSize(String size) {
        this.size = size;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public int getYear() {
        return year;
    }

    public void setYear(int year) {
        this.year = year;
    }

    public EquipmentType getEquipmentType() {
        return equipmentType;
    }

    public void setEquipmentType(EquipmentType equipmentType) {
        this.equipmentType = equipmentType;
    }

    public String getModel() {
        return this.model;
    }

    public void setModel(String model) {
        this.model = model;
    }

    public String getUrlImage() {
        return this.urlImage;
    }

    public void setUrlImage(String urlImage) {
        this.urlImage = urlImage;
    }

    @Override
    public String toString() {
        return "Equipment{" +
                "id=" + id +
                ", brand='" + brand +
                ", pricePerHour=" + pricePerHour +
                ", lenderInstructor=" + lenderInstructor +
                ", size='" + size +
                ", description='" + description +
                ", year=" + year +
                ", equipmentType=" + equipmentType +
                '}';
    }
}

File: MSI-project/src/app/models/equipment.ts
import { EquipmentType } from "./equipmentType";

export class Equipment {

  constructor(
    private _id: number,
    private _urlImage: string,
    private _brand: string,
    private _model: string,
    private _pricePerHour: number,
    private _lenderInstructor: string,
    private _size: string,
    private _description: string,
    private _year: string,
    private _equipmentType: EquipmentType){}

    public get year(): string {
      return this._year;
    }
    public set year(value: string) {
      this._year = value;
    }
    public get description(): string {
      return this._description;
    }
    public set description(value: string) {
      this._description = value;
    }

    public get model(): string {
      return this._model;
    }
    public set model(value: string) {
      this._model = value;
    }
    public get size(): string {
      return this._size;
    }
    public set size(value: string) {
      this._size = value;
    }

    public get urlImage(): string {
      return this._urlImage;
    }
    public set urlImage(value: string) {
      this._urlImage = value;
    }

    public get brand(): string {
      return this._brand;
    }
    public set brand(value: string) {
      this._brand = value;
    }
    public get lenderInstructor(): string {
      return this._lenderInstructor;
    }
    public set lenderInstructor(value: string) {
      this._lenderInstructor = value;
    }
    public get pricePerHour(): number {
      return this._pricePerHour;
    }
    public set pricePerHour(value: number) {
      this._pricePerHour = value;
    }
    public get id(): number {
      return this._id;
    }
    public set id(value: number) {
      this._id = value;
    }

    public get equipmentType(): EquipmentType {
      return this._equipmentType;
    }
    public set equipmentType(value: EquipmentType) {
      this._equipmentType = value;
    }

}
